docs/plugins/language.md

```diff
@@ -0,0 +1,207 @@
+
+```json
+{
+  "meta": {
+    "default": "en", // default fallback language
+    "names": { // human readable names for language codes
+        "en": "English",
+        "nl": "Nederlands"
+    },
+    "map": { // mapping language codes
+      "en-US": "en",
+      "en-GB": "en",
+      "nl-NL": "nl",
+      "nl-BE": "nl"
+    }
+  },
+  "en": {
+    //
+  },
+  "nl": {
+    //
+  }
+
+}
+```
+
+## Available methods
+
+### Set
+
+Sets or changes the current Language.
+
+```js
+const languageCode = 'en'
+Language.set(languageCode)
+```
+
+The language code should match with one of the language-keys specified in your `translations.json`.
+
+The `map` option in the `meta` key of your translations file can be used to specify _aliases_, and as such can be used
+to map different ISO codes (i.e. en-GB, en-US, etc.) to a single language key (en).
+
+If you have specified a `default` language in the `meta` key of your translations file, the Language plugin will use
+that as a fallback in case no matching language was found.
+
+Note, changing the language doesn't automatically trigger a template render update.
+
+#### Translate
+
+Translates a key to the matching value.
+
+```js
+Language.translate('hello')
+```
+
+The translate method will lookup the translation-key in the current selected language and return the _translated_ value.
+If no value is found in the dictionary of the current language, it will return the supplied key.
+
+Optionally you can specify _replacements_ in the translated string. There are 2 ways of defining replacements: _order_ based and _name_ based.
+
+Consider the following `translations.json`:
+
+```json
+{
+  "en": {
+    "introduction": "Hi, my name is {0}, my lastname is {1} and I am {2} years old",
+    "introduction2": "Hi, my name is {firstname}, my lastname is {lastname} and I am {age} years old",
+  }
+}
+```
+
+We can either pass a set of function argument for order based replacements.
+
+```js
+Language.translate('introduction1', 'John', 'Doe', 18)
+// Hi, my name is John, my lastname is Doe and I am 18 years old
+```
+
+Or a _key-value_-object for name based replacements.
+
+```js
+Language.translate('introduction2', {lastname: 'Smith', firstname: 'Joe', kids: 27}) // name based replacements
+// Hi, my name is Joe, my lastname is Smith and I am 27 years old
+```
+
+### Available
+
+Returns an Array with the available languages as specified in `translations.json`, which can for example be used to display
+a dynamic language switcher in your App.
+
+By default it will return a simple Array with the available language codes:
+
+```js
+Language.available() // ['en', 'nl', 'fr']
+```
+
+If you have specified an object with `names` in the `meta` key of your translations file, the `available`-method will return an
+Array of objects
+
+```js
+Language.available()
+// [{code: 'en', name: 'English', code: 'nl', name: 'Nederlands'}]
+```
```

Comment:
Mmmm, same function returning two different data structures. Are we sure about this?

Reply:
@wouterlucas I see your point ... the alternative would be to always construct an array of objects (with `code` and `name`) and set name to equal the code, when no names are specified in the metadata. That would make the return consistent. Do you prefer that?